rcgen/src/key_pair.rs

```diff
@@ -101,15 +105,44 @@ impl KeyPair {
 					serialized_der: key_pair_serialized,
 				})
 			},
+			#[cfg(all(feature = "aws_lc_rs", not(feature = "ring")))]
+			SignAlgo::Rsa(sign_alg) => self.generate_rsa_inner(KeySize::Rsa2048),
 			// Ring doesn't have RSA key generation yet:
 			// https://github.com/briansmith/ring/issues/219
 			// https://github.com/briansmith/ring/pull/733
-			// Nor does aws-lc-rs:
-			// https://github.com/aws/aws-lc-rs/issues/296
-			SignAlgo::Rsa() => Err(Error::KeyGenerationUnavailable),
+			#[cfg(any(not(feature = "aws_lc_rs"), feature = "ring"))]
+			SignAlgo::Rsa(_sign_alg) => Err(Error::KeyGenerationUnavailable),
 		}
 	}
 
+	/// Generates a new random RSA key pair for the specified key size
+	///
+	/// If passed a signature algorithm that is not RSA, it will return
+	/// [`Error::KeyGenerationUnavailable`].
+	#[cfg(all(feature = "crypto", feature = "aws_lc_rs", not(feature = "ring")))]
+	pub fn generate_rsa_for(alg: &'static SignatureAlgorithm) -> Result<Self, Error> {
```

Comment:
What is this method for? I don't think it could compile?

Reply:
Ah yeah sorry, I have pushed an unfinished state. I have fixed it now. The method is to allow customizing the key size (which defaults to 2048).